test/test_prims.py

```diff
@@ -113,6 +113,122 @@ def _wrapper(a):
             self.assertTrue(result.is_contiguous)
             self.assertEqual(_wrapper(a), result)
 
+    @onlyCUDA
+    @skipCUDAIfRocm
+    @dtypes(torch.float32)
+    def test_copy_to(self, device, dtype):
+        from torch.fx.experimental.proxy_tensor import make_fx
+        from torch._prims.context import TorchRefsNvfuserCapabilityMode
+        from torch._prims.executor import execute
+
+        def func1(a, b):
+            return (a.copy_(b),)
+
+        def func2(a, b):
+            b_sin = b.sin()
+            return (a.copy_(b_sin),)
+
+        def func3(a, b):
+            return (a.copy_(b), b)
+
+        def func4(a, b):
+            b_sin = b.sin()
+            return (a.copy_(b_sin), b_sin)
+
+        def func5(a, b):
+            b_sin = b.sin()
+            a1 = a.copy_(b_sin)
+            a1_sin = a1.sin()
+            a2 = a1.copy_(a1_sin)
```

Comment:
This looks scary.

Currently codegen has no idea on proper race condition. Having a single buffer used as both RW in different operation is not something we support. We should remove this test.

Reply:
Removing test is easy, but what happens here? Does codegen generate a kernel for this fusion, or does it fallback?

Reply:
Yes, the codegen generates kernels for this fusion. There are no fallbacks at the nvFuser level, it's either more segments or an error. Since we're using `executor="strictly_nvfuser"` there's no fallback at the Python level as well.

<details>
<summary>Codegen generates two segments in this case:</summary>

```cpp
Segmented_Fusion Dump: -- fusion segments:
Segmented_Fusion{
groups:
g{0, 1}

g{2, 3}

edges:

group details:
g{(pointwise)
inputs:
T0_g[ iS0{i0}, iS1{i1} ] float
T1_g[ iS2{i3}, iS3{i4} ] float
outputs:
T2_g[ iS4{i3}, iS5{i4} ] float
T3_g[ iS6{i3}, iS7{i4} ] float


T2_g[ iS4{i3}, iS5{i4} ]
   = sinf(T1_g[ iS2{i3}, iS3{i4} ]);
T3_g[ iS6{i3}, iS7{i4} ]
   = T2_g[ iS4{i3}, iS5{i4} ];
}

g{(pointwise)
inputs:
T0_g[ iS0{i0}, iS1{i1} ] float
outputs:
T4_g[ iS8{i0}, iS9{i1} ] float
T5_g[ iS10{i0}, iS11{i1} ] float


T4_g[ iS8{i0}, iS9{i1} ]
   = sinf(T0_g[ iS0{i0}, iS1{i1} ]);
T5_g[ iS10{i0}, iS11{i1} ]
   = T4_g[ iS8{i0}, iS9{i1} ];
}

} //Segmented_Fusion
```
</details>

<details>
<summary>This corresponds to two kernels:</summary>

```cpp
======= Codegen output for kernel: kernel1 =======

__global__ void kernel1(Tensor<float, 2> T1, Tensor<float, 2> T0, Tensor<float, 2> T7, Tensor<float, 2> T3) {
  int i71;
  i71 = (((nvfuser_index_t)blockIdx.x) * 128) + ((nvfuser_index_t)threadIdx.x);
  if ((i71 < (T1.size[0] * T1.size[1]))) {
    float T6[1];
    T6[0] = 0;
    T6[0]
       = T1[i71];
    float T2[1];
    T2[0]
       = sinf(T6[0]);
    float T8[1];
    T8[0]
       = T2[0];
    T7[i71]
       = T8[0];
    float T9[1];
    T9[0]
       = T2[0];
    T3[i71]
       = T9[0];
  }
}

======================================


======= Codegen output for kernel: kernel2 =======

__global__ void kernel2(Tensor<float, 2> T0, Tensor<float, 2> T7, Tensor<float, 2> T5) {
  int i69;
  i69 = (((nvfuser_index_t)blockIdx.x) * 128) + ((nvfuser_index_t)threadIdx.x);
  if ((i69 < (T0.size[0] * T0.size[1]))) {
    float T6[1];
    T6[0] = 0;
    T6[0]
       = T0[i69];
    float T4[1];
    T4[0]
       = sinf(T6[0]);
    float T8[1];
    T8[0]
       = T4[0];
    T7[i69]
       = T8[0];
    float T9[1];
    T9[0]
       = T4[0];
    T5[i69]
       = T9[0];
  }
}

======================================
```
</details>